src/linker/configuration.py

```diff
@@ -83,6 +82,22 @@ def spark_resources(self) -> Dict[str, Any]:
             **{k: v for k, v in self.spark.items() if k != "workers"},
         }
 
+    def get_implementation_specific_configuration(self, step_name: str) -> Dict[str, Any]:
+        """Extracts and formats an implementation-specific configuration from the pipeline config"""
+
+        def _stringify_keys_values(config: Any) -> Dict[str, Any]:
+            # Singularity requires env variables be strings
+            if isinstance(config, Dict):
+                return {
+                    str(key): _stringify_keys_values(value) for key, value in config.items()
+                }
+            else:
+                # The last step of the recursion is not a dict but the leaf node's value
+                return str(config)
+
+        config = self.pipeline["steps"][step_name]["implementation"].get("configuration", {})
+        return _stringify_keys_values(config)
+
     def copy_configuration_files_to_results_directory(self, results_dir: Path) -> None:
```

Comment:
Why is this still here? Is this branch out of date?

Reply:
whoops, good catch. Evidently I copy/pasted instead of cut/pasted (though I was correctly importing it from data_utils)